go.mod

```diff
@@ -5,39 +5,58 @@ go 1.15
+	github.com/containerd/containerd v1.4.4 // indirect
+	github.com/containerd/stargz-snapshotter/estargz v0.5.0 // indirect
+	github.com/docker/cli v20.10.5+incompatible // indirect
+	github.com/docker/docker v20.10.5+incompatible // indirect
 	github.com/ghodss/yaml v1.0.0
 	github.com/go-git/go-git-fixtures v3.5.0+incompatible
 	github.com/go-git/go-git/v5 v5.1.0
 	github.com/go-openapi/spec v0.19.9
-	github.com/google/go-cmp v0.5.1
-	github.com/google/go-containerregistry v0.1.1
-	github.com/gophercloud/gophercloud v0.4.0 // indirect
+	github.com/gogo/protobuf v1.3.2 // indirect
+	github.com/golang/protobuf v1.5.1 // indirect
+	github.com/google/go-cmp v0.5.5
+	github.com/google/go-containerregistry v0.4.1
+	github.com/google/go-containerregistry/pkg/authn/k8schain v0.0.0-20210322164748-a11b12f378b5
+	github.com/jinzhu/gorm v1.9.12 // indirect
+	github.com/lib/pq v1.2.0 // indirect
 	github.com/matthewmcnew/archtest v0.0.0-20191014222827-a111193b50ad
+	github.com/mattn/go-colorable v0.1.8 // indirect
 	github.com/mgutz/ansi v0.0.0-20170206155736-9520e82c474b
+	github.com/mitchellh/mapstructure v1.3.1 // indirect
+	github.com/moby/term v0.0.0-20201216013528-df9cb8a40635 // indirect
+	github.com/pelletier/go-toml v1.8.0 // indirect
 	github.com/pkg/errors v0.9.1
 	github.com/sclevine/spec v1.4.0
+	github.com/sirupsen/logrus v1.8.1 // indirect
+	github.com/spf13/cast v1.3.1 // indirect
+	github.com/spf13/jwalterweatherman v1.1.0 // indirect
 	github.com/spf13/pflag v1.0.5
+	github.com/spf13/viper v1.7.0 // indirect
 	github.com/stretchr/testify v1.6.1
 	github.com/theupdateframework/notary v0.6.2-0.20200804143915-84287fd8df4f
-	github.com/vdemeester/k8s-pkg-credentialprovider v1.17.4
-	go.uber.org/zap v1.15.0
-	golang.org/x/crypto v0.0.0-20200510223506-06a226fb4e37
-	golang.org/x/sync v0.0.0-20200317015054-43a5402ce75a
-	k8s.io/api v0.17.6
-	k8s.io/apimachinery v0.17.6
+	github.com/vdemeester/k8s-pkg-credentialprovider v1.19.7
+	go.uber.org/zap v1.16.0
+	golang.org/x/crypto v0.0.0-20201016220609-9e8e0b390897
+	golang.org/x/net v0.0.0-20210316092652-d523dce5a7f4 // indirect
+	golang.org/x/sync v0.0.0-20210220032951-036812b2e83c
+	golang.org/x/sys v0.0.0-20210320140829-1e4c9ba3b0c4 // indirect
+	google.golang.org/genproto v0.0.0-20210322173543-5f0e89347f5a // indirect
+	gopkg.in/ini.v1 v1.56.0 // indirect
+	gopkg.in/src-d/go-git.v4 v4.13.1 // indirect
+	k8s.io/api v0.19.7
+	k8s.io/apimachinery v0.19.7
 	k8s.io/client-go v11.0.1-0.20190805182717-6502b5e7b1b5+incompatible
-	k8s.io/code-generator v0.18.6
-	k8s.io/kube-openapi v0.0.0-20200410145947-bcb3869e6f29
-	knative.dev/pkg v0.0.0-20200702222342-ea4d6e985ba0
+	k8s.io/code-generator v0.19.7
+	k8s.io/kube-openapi v0.0.0-20210113233702-8566a335510f
+	knative.dev/pkg v0.0.0-20210323202917-b558677ab034
 )
 
-replace (
-	k8s.io/client-go => k8s.io/client-go v0.17.6
-	k8s.io/code-generator => k8s.io/code-generator v0.17.6
-)
+replace k8s.io/client-go => k8s.io/client-go v0.19.7
```

Comment:
Can we avoid the replace directive now?

Reply:
we found that go mod won't reconcile to the right version without it